### PR TITLE
Adding documentation to util/logging

### PR DESCRIPTION
**Pull request by nyarly**

### CODE REVIEW

**Comment by sous-machine-account** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: util/logging/callerinfo.go [hunk 1]
```diff
@@ -36,6 +36,10 @@ type (
 var notGCI excludedFrame
 
 // GetCallerInfo captures a CallerInfo based on where it's called.
+// It's very common to call this like:
+//   GetCallerInfo(logging.NotHere())
```
Comment (samsalisbury) [inline util/logging/callerinfo.go]: Nice, new one to me that you can do godoc code blocks without extra newlines.

File: util/logging/logging.go [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+// The design of this package is premised on using the Go type system
+// to tighten the iteration cycle around working with our ELK stack.
+// Specifically, we define explicit log message structs,
+// and use constructor functions in order to control exactly the
+// fields that have to be collected. Ideally, on-the-spot log messages
+// will be guided by their constructors to provide the correct fields,
+// so that we're shielded from the dead-letter queue at the time of
+// logging.
+//
+// As a compliment to this process, we also allow for log messages
```
Comment (samsalisbury) [inline util/logging/logging.go]: By implementing any combination of the interfaces `LogMessage`, `ConsoleMessage` and `MetricsMessage` single log message may emit:

- structured logs to the ELK stack (`LogMessage`)
- textual logs to stderr (`ConsoleMessage`), or
- metrics messages to graphite carbon endpoints (`MetricsMessage`)

Each message must implement at least one of those interfaces.

File: util/logging/logging.go [hunk 3]
```diff
@@ -50,27 +116,26 @@ type (
...
-	// Notice that the global LotSet doesn't have metrics available - when you
-	// want metrics in a component, you need to add an injected LogSet. c.f.
-	// ext/docker/image_mapping.go
-	Log = func() LogSet {
-		//return *(NewLogSet(semv.MustParse("0.0.0"), "sous.global", "", os.Stderr))
-		return *(SilentLogSet()) // we'll configure it later
-	}()
-)
+// Log collects various loggers to use for different levels of logging
+// Deprecation warning: the global logger is slated for removal.
```
Comment (samsalisbury) [inline util/logging/logging.go]: If we begin this line with `Deprecated: ` GoDoc will treat Log specially as deprecated, see https://blog.golang.org/godoc-documenting-go-code

Comment (nyarly) [reply util/logging/logging.go]: Sadly, go vet (or one of the linters) also considers the use of Deprecated: members as an error. Worth discussing how to address - e.g. we could configure that as a check we don't care about or else touch all the cases and add an exclusion (if the linter even allows for that)

**Comment by sous-machine-account** [comment]
LGTM

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**